sdks/python/apache_beam/runners/worker/opcounters.py

```diff
@@ -24,11 +24,99 @@
 import math
 import random
 
+from apache_beam.utils import counters
 from apache_beam.utils.counters import Counter
+from apache_beam.utils.counters import CounterName
 
 # This module is experimental. No backwards-compatibility guarantees.
 
 
+class TransformIoCounter(object):
+  """Class to track time and bytes consumed while reading from IO.
+
+  Examples of IO can be side inputs, shuffle, or streaming state.
+  """
+
+  def check_step(self):
+    pass
+
+  def __exit__(self, exc_type, exc_value, traceback):
+    pass
+
+  def __enter__(self):
+    pass
+
+  def add_bytes_read(self, n):
+    pass
+
+
+class SideInputReadCounter(TransformIoCounter):
+  """Tracks time and bytes consumed while reading from side inputs.
+
+  This class is designed to track consumption of side inputs across fused steps.
+  We represent a side input as a declaring step, and an input index, which are
+  represented as an io_target.
```

Comment:
<!--thread_id:cc_158386099_t; commit:1e64e1f8b880a9fe4450c84cf1c7c5694eb3567e; resolved:0-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> Do we define what we mean by an `io_target` elsewhere?
> 
> Please add extra newline after paragraph.

<!--section:body-->
There's a small comment in counters.py about it. I can improve that if you think it's fitting.